src/views/connectionUI.ts

```diff
@@ -85,8 +85,10 @@ export class ConnectionUI {
 	 * @returns The connectionInfo choosen or created from the user, or undefined if the user cancels the prompt.
 	 */
 	public async promptForConnection(ignoreFocusOut: boolean = false): Promise<IConnectionInfo | undefined> {
-		return await new Promise<IConnectionInfo | undefined>((resolve, _) => {
-			let connectionProfileList = this._connectionStore.getPickListItems();
+		// Let this design use Promise and resolve/reject pattern instead of async/await
+		// because resolve/reject is done in in callback events.
+		return await new Promise<IConnectionInfo | undefined>(async (resolve, _) => {
```

Comment:
You also need to remove the await here. This is going to force the function to wait until the promise resolves (which means the user has picked an item) which is breaking the test as well.

Reply:
That had been there before as well..

Reply:
Oh yeah, don't know what I was thinking here. It would only be a problem if the test awaited the returned promise first. So yeah nevermind.

(although personally I don't like "return await" - it's just more verbose and unnecessarily unwrapping and re-wrapping the returned value. The only reason you'd want to do it like this is to keep the callstack exactly correct but that isn't something I've ever seen as being super necessary)